nifi-nar-bundles/nifi-hubspot-bundle/nifi-hubspot-processors/src/main/java/org/apache/nifi/processors/hubspot/GetHubSpot.java

```diff
@@ -187,78 +257,158 @@ private String getResponseBodyAsString(final ProcessContext context, final HttpR
+        try {
+            return webClientServiceProvider.getWebClientService()
+                    .post()
+                    .uri(uri)
+                    .header("Authorization", "Bearer " + accessToken)
+                    .header("Content-Type", "application/json")
+                    .body(inputStream, OptionalLong.of(inputStream.available()))
+                    .retrieve();
+        } catch (IOException e) {
+            throw new ProcessException("Could not transform incremental filters to input stream", e);
+        }
     }
 
-    private URI createUri(final ProcessContext context, final StateMap state) {
-        final String path = context.getProperty(OBJECT_TYPE).getValue();
-        final HttpUriBuilder uriBuilder = getBaseUri(context);
+    private String createIncrementalFilters(final ProcessContext context, final Map<String, String> stateMap) {
+        final String limit = context.getProperty(RESULT_LIMIT).getValue();
+        final String objectType = context.getProperty(OBJECT_TYPE).getValue();
+        final HubSpotObjectType hubSpotObjectType = OBJECT_TYPE_LOOKUP_MAP.get(objectType);
+        final Long incrDelayMs = context.getProperty(INCREMENTAL_DELAY).asTimePeriod(TimeUnit.MILLISECONDS);
+        final String cursorKey = String.format(CURSOR_KEY_PATTERN, objectType);
 
-        final boolean isLimitSet = context.getProperty(RESULT_LIMIT).evaluateAttributeExpressions().isSet();
-        if (isLimitSet) {
-            final String limit = context.getProperty(RESULT_LIMIT).getValue();
-            uriBuilder.addQueryParameter(LIMIT_PARAMETER, limit);
+        final ObjectNode root = OBJECT_MAPPER.createObjectNode();
+        root.put("limit", limit);
+
+        final String cursor = stateMap.get(cursorKey);
+        if (cursor != null && !NO_PAGING.equals(cursor)) {
+            root.put(PAGING_CURSOR, stateMap.get(cursorKey));
         }
+        final boolean isIncremental = context.getProperty(IS_INCREMENTAL).asBoolean();
+        if (isIncremental) {
+            final String initialStartTimeValue = context.getProperty(INCREMENTAL_INITIAL_START_TIME).getValue();
+            String initialStartTimeEpoch = getInitialStartTimeEpoch(initialStartTimeValue);
+            final String hubspotSpecificIncrementalFieldName = hubSpotObjectType.getLastModifiedDateType().getValue();
+            final String lastStartTime = stateMap.get(START_INCREMENTAL_KEY);
+            final String lastEndTime = stateMap.getOrDefault(END_INCREMENTAL_KEY, initialStartTimeEpoch);
+
+            final String currentStartTime;
+            final String currentEndTime;
+
+            if (cursor != null && !NO_PAGING.equals(cursor)) {
+                currentStartTime = lastStartTime;
+                // lastEndTime can be null if incremental loading was turned off beforehand
+                currentEndTime = lastEndTime != null ? lastEndTime : String.valueOf(getCurrentEpochTime());
```

Comment:
I think it is better to avoid this situation and the state should also be reset when the user changes `Incremental Loading` (similar to` Object Type`).
In that case `currentEndTime = lastEndTime` would be enough because `lastEndTime` must have a value when the paging cursor is set.
Otherwise `Incremental Delay` should be applied here too.